spec/solver_spec.rb

```diff
@@ -0,0 +1,34 @@
+require './solver'
+
+describe Solver do
+  it 'should return 1 for 0' do
+    Solver.new.factorial(0).should == 1
+  end
+  it 'should return 120 for 5' do
+    Solver.new.factorial(5).should == 120
+  end
```

Comment:
- [ ] Kindly make sure the `#factorial` method has tests that handle negative value(s) which raises an exception as an expectation as required for the project.

Reply:
Thank you @tresorsawasawa for your feedback, I will add raises exception test.